test/MusicStore.Test/ManageControllerTest.cs

```diff
@@ -0,0 +1,116 @@
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Core;
+using Microsoft.AspNet.Http.Core.Security;
+using Microsoft.AspNet.Http.Security;
+using Microsoft.AspNet.Identity;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.DependencyInjection.Fallback;
+using Xunit;
+using MusicStore.Models;
+
+namespace MusicStore.Controllers
+{
+    public class ManageControllerTest
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public ManageControllerTest()
+        {
+            var services = new ServiceCollection();
+            services.AddEntityFramework()
+                    .AddInMemoryStore()
+                    .AddDbContext<MusicStoreContext>();
+
+            services.AddIdentity<ApplicationUser, IdentityRole>()
+                    .AddEntityFrameworkStores<MusicStoreContext>();
+
+            // IHttpContextAccessor is required for SignInManager, and UserManager
+            services.AddInstance<IHttpContextAccessor>(new TestHttpContextAcccessor());
+
+            _serviceProvider = services.BuildServiceProvider();
+        }
+
+        [Fact]
+        public async Task Index_ReturnsViewBagMessagesExpected()
+        {
+            // Arrange
+            var userId = "TestUserA";
+            var phone = "abcdefg";
+            var claims = new List<Claim> { new Claim(ClaimTypes.NameIdentifier, userId) };
+
+            var userManager = _serviceProvider.GetRequiredService<UserManager<ApplicationUser>>();
+            var userManagerResult = await userManager.CreateAsync(
+                new ApplicationUser { Id = userId, UserName = "Test", TwoFactorEnabled = true, PhoneNumber = phone },
+                "Pass@word1");
+            Assert.True(userManagerResult.Succeeded);
+
+            var signInManager = _serviceProvider.GetRequiredService<SignInManager<ApplicationUser>>();
+
+            var httpContext = _serviceProvider.GetRequiredService<IHttpContextAccessor>().Value;
+            httpContext.User = new ClaimsPrincipal(new ClaimsIdentity(claims));
+
+            var controller = new ManageController()
+            {
+                ActionContext = new ActionContext(
+                    httpContext,
+                    new RouteData(),
+                    new ActionDescriptor()),
+                UserManager = userManager,
+                SignInManager = signInManager,
+            };
+
+            // Act
+            var result = await controller.Index();
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.NotNull(viewResult.ViewData);
+
+            var model = Assert.IsType<IndexViewModel>(viewResult.ViewData.Model);
+            Assert.True(model.TwoFactor);
+            Assert.Equal(phone, model.PhoneNumber);
+            Assert.True(model.HasPassword);
+        }
+
+        private class TestHttpContextAcccessor : IHttpContextAccessor
```

Comment:
Though BTW I think we also just renamed the `Value` property to be `Context` or something so keep an eye on that before merging.